pythran/pythonic/types/bool.hpp

```diff
@@ -3,4 +3,35 @@
 
 #include "pythonic/include/types/bool.hpp"
 
+#ifdef ENABLE_PYTHON_MODULE
+
+#include "pythonic/python/core.hpp"
+
+namespace pythonic
+{
+  template <>
+  struct to_python<bool> {
+    static PyObject *convert(bool b)
+    {
+      if (b)
+        Py_RETURN_TRUE;
+      else
+        Py_RETURN_FALSE;
+    }
+  };
+  template <>
+  struct from_python<bool> {
+    static bool is_convertible(PyObject *obj)
+    {
+      return PyBool_Check(obj);
+    }
+    static bool convert(PyObject *obj)
+    {
+      return PyInt_AsLong(obj);
+    }
+  };
+}
+
+#endif
+
```

Comment:
This won't be an issue. The idea is that I export (in a pythonic/include/.../...hpp) the ``to_python()`` function and it gets specialized in the relevant header. If you want to call a specific ``to_python`` specialization, just call the good one and it will be automagically dispatched thanks to the template specialization mechanism. The only problem that could appear would be template use before specialization. We just need to respect the correct include order to by pass this.

The benfit ? cleaner include files as only the interface is exposed there. For instance sometimes the ``namespace impl`` stuff are also exported, which should be avoided (that's how I understand the ``impl`` namespace.)